pkg/local_object_storage/blobstor/peapod/peapod.go

```diff
@@ -492,3 +494,31 @@ func (x *Peapod) Iterate(prm common.IteratePrm) (common.IterateRes, error) {
 
 	return common.IterateRes{}, nil
 }
+
+// IterateAddresses iterates over all objects stored in the underlying database
+// and passes their addresses into f. If f returns an error, IterateAddresses
+// returns it and breaks.
+func (x *Peapod) IterateAddresses(f func(addr oid.Address) error) error {
+	var addr oid.Address
+
+	err := x.bolt.View(func(tx *bbolt.Tx) error {
+		bktRoot := tx.Bucket(rootBucket)
+		if bktRoot == nil {
+			return errMissingRootBucket
+		}
+
+		return bktRoot.ForEach(func(k, v []byte) error {
```

Comment:
IIRC `v` here just points to some mapped area, so it doesn't induce any real overhead (copying/decoding).